kratos/processes/entity_erase_process.cpp

```diff
@@ -0,0 +1,162 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/kratos_flags.h"
+#include "processes/entity_erase_process.h"
+#include "utilities/variable_utils.h"
+#include "utilities/auxiliar_model_part_utilities.h"
+
+namespace Kratos
+{
+template<class TEntity>
+const Kratos::Flags EntitiesEraseProcess<TEntity>::ERASE_ALL_ENTITIES(Kratos::Flags::Create(0));
+template<class TEntity>
+const Kratos::Flags EntitiesEraseProcess<TEntity>::NOT_ERASE_ALL_ENTITIES(Kratos::Flags::Create(0, false));
+template<class TEntity>
+const Kratos::Flags EntitiesEraseProcess<TEntity>::ASSIGN_FLAG(Kratos::Flags::Create(1));
+template<class TEntity>
+const Kratos::Flags EntitiesEraseProcess<TEntity>::NOT_ASSIGN_FLAG(Kratos::Flags::Create(1, false));
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TEntity>
+EntitiesEraseProcess<TEntity>::EntitiesEraseProcess(
+    ModelPart& rModelPart,
+    Flags Options
+    )
+    : mrModelPart(rModelPart),
+     mrOptions(Options)
+{
+    KRATOS_TRY
+
+    KRATOS_CATCH("")
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TEntity>
+EntitiesEraseProcess<TEntity>::EntitiesEraseProcess(
+    ModelPart& rModelPart,
+    Parameters ThisParameters
+    )
+    : mrModelPart(rModelPart)
+{
+    KRATOS_TRY
+
+    // The default parameters
+    Parameters default_parameters = GetDefaultParameters();
+    ThisParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+    // Assign parameters
+    mrOptions.Set(ERASE_ALL_ENTITIES, ThisParameters["remove_from_all_levels"].GetBool());
+    mrOptions.Set(ASSIGN_FLAG, ThisParameters["assign_flag"].GetBool());
+
+    KRATOS_CATCH("")
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+void EntitiesEraseProcess<Node<3>>::Execute()
+{
+    KRATOS_TRY;
+
+    // If we assign flags
+    if (mrOptions.Is(ASSIGN_FLAG)) VariableUtils().SetFlag(TO_ERASE, true, mrModelPart.Nodes());
```

Comment:
@pooyan-dadvand here the flag is assigned, and later removed, the idea is that you can assign the flag from outside the process

Reply:
You should do this when the `ERASE_ALL_ENTITIES` is set and you don't need this additional flag...

Reply:
But the flag doesn't indicate that all entities must be removed, it indicates is removed in all levels (that's was why the previous flag was called that way "REMOVE_FROM_ALL_LEVELS")

Reply:
This is another flag.... I think you are mixing different ones...

The  `ERASE_ALL_ENTITIES` would remove all entities from this model_part. What you are doing with assign flag without being descriptive.